compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -9076,6 +9075,9 @@ private LinkedHashSet<BType> collectTupleFieldTypes(BTupleType tupleType, Linked
                         memberTypes.add(memberType);
                     }
                 });
+        if (tupleType.restType != null) {
```

Comment:
We can extract `tupleType.restType` out to a variable.

Reply:
Updated. 1ef87477be4270f73cec77393ed6e541fc0886eb